### PR TITLE
Use LIBSVM in SVM example

### PR DESCRIPTION
**Pull request by willtebbutt**
As discussed previously, our SVM example wasn't actually an SVM.

This PR utilises LIBSVM.jl to make it actually be an SVM.

### CODE REVIEW

**Comment by theogf** [comment]
That's nice! At some point let's try to see if we can get an implementation with [SimpleSVMs.jl](https://github.com/matbesancon/SimpleSVMs.jl) @matbesancon

**Comment by willtebbutt** [comment]
Glad you approve @theogf . Are you happy to approve this in the mean time?

File: examples/support-vector-machine/script.jl [hunk 1]
```diff
@@ -20,23 +18,29 @@ Random.seed!(1234);
...
-Xgrid = ColVecs(mapreduce(collect, hcat, Iterators.product(xgrid, xgrid)));
+test_range = range(floor(Int, minimum(X)), ceil(Int, maximum(X)); length=100)
+x_test = ColVecs(mapreduce(collect, hcat, Iterators.product(test_range, test_range)));
 
 # Create kernel function:
 k = SqExponentialKernel() ∘ ScaleTransform(2.0)
 
-# Optimal prediction:
-f(x, X, k, λ) = kernelmatrix(k, x, X) / (kernelmatrix(k, X) + exp(λ) * I) * y
+# LibSVM can make use of a pre-computed kernel matrix.
```
Comment (theogf) [inline examples/support-vector-machine/script.jl:36]:
```suggestion
# [LIBSVM](https://github.com/JuliaML/LIBSVM.jl) can make use of a pre-computed kernel matrix.
```

Comment (willtebbutt) [reply examples/support-vector-machine/script.jl:36]: Good idea

**Review by theogf — APPROVED** [review]

**Comment by matbesancon** [comment]
yes we could/should plumb the two together at some point

File: examples/support-vector-machine/script.jl [hunk 2]
```diff
@@ -20,23 +18,29 @@ Random.seed!(1234);
...
 
-# Optimal prediction:
-f(x, X, k, λ) = kernelmatrix(k, x, X) / (kernelmatrix(k, X) + exp(λ) * I) * y
+# [LIBSVM](https://github.com/JuliaML/LIBSVM.jl) can make use of a pre-computed kernel matrix.
+# KernelFunctions.jl can be used to produce that.
+# Precomputed matrix for training (corresponds to linear kernel)
+model = svmtrain(kernelmatrix(k, x_train), y_train; kernel=LIBSVM.Kernel.Precomputed)
+
+# Precomputed matrix for prediction
+y_pr, _ = svmpredict(model, kernelmatrix(k, x_train, x_test))
```
Comment (st--) [inline examples/support-vector-machine/script.jl:42]:
```suggestion
y_pr, _ = svmpredict(model, kernelmatrix(k, x_train, x_test));
```

File: examples/support-vector-machine/script.jl [hunk 3]
```diff
@@ -20,23 +18,29 @@ Random.seed!(1234);
...
-y = rand([-1, 1], N);
+y_train = rand([-1, 1], N);
 
 # Random attributes for both classes:
 X = Matrix{Float64}(undef, 2, N)
-rand!(MvNormal(randn(2), I), view(X, :, y .== 1))
-rand!(MvNormal(randn(2), I), view(X, :, y .== -1));
+rand!(MvNormal(randn(2), I), view(X, :, y_train .== 1))
+rand!(MvNormal(randn(2), I), view(X, :, y_train .== -1));
+x_train = ColVecs(X)
```
Comment (st--) [inline examples/support-vector-machine/script.jl:27]:
```suggestion
x_train = ColVecs(X);
```

**Review by st-- — APPROVED** [review]

**Pull request merged**